vertx-client/src/main/java/org/apache/cassandra/sidecar/client/VertxRequestExecutor.java

```diff
@@ -71,4 +75,18 @@ public void close() throws Exception
             logger.warn("Failed to close vertx after 1 minute", exception);
         }
     }
+
+    @Override
+    void applyRetryPolicy(CompletableFuture<HttpResponse> future, Iterator<SidecarInstance> iterator, SidecarInstance sidecarInstance, RequestContext context, int attempt, HttpResponse response, Throwable throwable)
+    {
+        if (context.request() instanceof DownloadableRequest && throwable instanceof FileSystemException)
+        {
+            // If the request is a downloadable request and the exception is a FileSystemException,
+            // then it is an issue with local file system. In this case, we should not retry the request.
+            future.completeExceptionally(throwable);
+            return;
+        }
+
+        super.applyRetryPolicy(future, iterator, sidecarInstance, context, attempt, response, throwable);
+    }
```

Comment:
Is it specific to live migration? If so, would you like to create a live-migration-specific retry policy instead?